server/routes/orders.js

```diff
@@ -1,90 +1,275 @@
+    if (admin || notUser) {
+      const order = await Order.create(req.body);
+      res.status(201).send(order);
+    } else {
+      res.sendStatus(403);
     }
-    catch(err) {
-        console.log(err);
+  } catch (err) {
+    console.log(err);
+  }
+});
+
+orderRoute.delete("/:id", async (req, res, next) => {
+  try {
+    const admin = req.user && req.user.class === "admin";
+    const ownUser = req.user.id === req.params.id * 1;
+    if (admin || ownUser) {
+      await Order.destroy({ where: { id: req.params.id } });
+      res.sendStatus(204);
+    } else {
+      res.sendStatus(403);
     }
-})
-
-orderRoute.put('/:id', async(req, res, next) => {
-    try {
-        const admin = req.user && req.user.class === 'admin';
-        const ownUser = req.user.id === req.params.id * 1;
-        if (admin || ownUser) {
-            const order = await Order.findByPk(req.params.id);
-            await Order.update(req.body);
-            res.send(order);
-        } else {
-            res.sendStatus(403);
-        }
+  } catch (err) {
+    console.log(err);
+  }
+});
+
+orderRoute.put("/:id", async (req, res, next) => {
+  try {
+    const admin = req.user && req.user.class === "admin";
+    const ownUser = req.user.id === req.params.id * 1;
+    if (admin || ownUser) {
+      const order = await Order.findByPk(req.params.id, {
+        include: [{ model: Item }, { model: User }],
+      });
+      await Order.update(req.body, { where: { id: req.params.id } });
+      res.send(order);
+    } else {
+      res.sendStatus(403);
     }
-    catch(err) {
-        console.log(err);
+  } catch (err) {
+    console.log(err);
+  }
+});
+
+orderRoute.post("/makeOrder", async (req, res, next) => {
+  try {
+    const { token, cart, user } = req.body;
+    const price = req.body.cart.items.reduce((acc, item) => {
+      acc += item.orderItem.priceOrdered * 1 * item.orderItem.quantity;
+      return acc;
+    }, 0);
+    const customer = await stripe.customers.create({
+      email: token.email,
+      source: token.id,
+    });
+    const uniqueID = uuid();
+
+    await stripe.charges.create(
+      {
+        amount: price * 100,
+        currency: "usd",
+        customer: customer.id,
+        receipt_email: token.email,
+        description: "Confirmation",
+      },
+      {
+        idempotencyKey: uniqueID,
+      }
+    );
+
+    const gotOrder = await Order.findByPk(cart.id);
+    gotOrder.status = "ordered";
+    await gotOrder.save();
+
+    const gotUser = await User.findByPk(user.id);
+    const newCart = await Order.create({ status: "cart" });
+    await gotUser.addOrder(newCart);
+
+    cart.items.map(async (item) => {
+      const foundItem = await Item.findByPk(item.id);
+      foundItem.inventory -= item.orderItem.quantity * 1;
+      await foundItem.save();
+    });
+
+    const cartToSend = await Order.findByPk(newCart.id, {
+      include: [
+        {
+          model: Item,
+        },
+      ],
+    });
```

Comment:
Food for thought: Making this whole thing a SQL transaction for rollback guarantees.